### PR TITLE
Make the Airbrake plugin useable

### PR DESCRIPTION
**Pull request by moltenice**
## Hey, I just made a Pull Request!

This will be the PR that makes the Airbrake plugin useable in Backstage.

- Airbrake frontend and backend work together. Previously they were developed separately so the frontend wasn't setup to talk to the backend.
- README has been updated to provide loads of documentation on installation and also when developing the plugins.
- The config is optional in development so the plugin being present doesn't hinder development on this repo. As a result, I have reverted @Rugvip's PR here: https://github.com/backstage/backstage/pull/9601.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by punitcse — APPROVED** [review]
🍰

File: plugins/airbrake-backend/src/config/ExtractAirbrakeConfig.ts [hunk 1]
```diff
@@ -33,9 +34,27 @@ export interface AirbrakeConfig {
...
-  return {
-    apiKey: config.getString('airbrake.apiKey'),
-  };
+export function extractAirbrakeConfig(
+  config: Config,
+  logger: winston.Logger,
+): AirbrakeConfig {
+  try {
+    return {
+      apiKey: config.getString('airbrake.apiKey'),
```
Comment (freben) [inline plugins/airbrake-backend/src/config/ExtractAirbrakeConfig.ts:45]: I was going to suggest that you could use getOptionalString here instead and check if undefined comes out - but I guess that the way you do it here actually leverages the error message that comes out of the config system sensibly :)

File: plugins/airbrake/package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
   "dependencies": {
     "@backstage/catalog-model": "^0.10.0",
     "@backstage/core-components": "^0.8.9",
+    "@backstage/core-app-api": "^0.5.3",
```
Comment (freben) [inline plugins/airbrake/package.json:28]: Hm! This one should not be needed at all!

**Review by freben — COMMENTED** [review]
Thanks! Just a few smaller notes

File: packages/backend/package.json [hunk 3]
```diff
@@ -33,6 +33,7 @@
     "@backstage/catalog-model": "^0.10.0",
     "@backstage/config": "^0.1.14",
     "@backstage/integration": "^0.7.3",
+    "@backstage/plugin-airbrake-backend": "^0.1.0",
```
[the remark this answers is not in the document]
Comment (moltenice) [reply packages/backend/package.json]: Removed :)

File: plugins/airbrake/package.json (hunk 2 above)
Comment (moltenice) [reply plugins/airbrake/package.json]: Removed :)

Comment (freben) [reply plugins/airbrake/package.json:28]: Ah, but then I'd prefer that you actually fake it instead - just implement the interface with its single method

```ts
const fake = {
  getBaseUrl(t: string) {
    return `http://localhost:7007/api/${encodeURIComponent(t)}`
  }
} 
```

or so

Comment (freben) [reply plugins/airbrake/package.json:28]: oh no, hang on, you are initializing this mock from somewhere right - you may have been able to grab the actual discoveryApiRef at that point and injecting it right?

Comment (moltenice) [reply plugins/airbrake/package.json]: @freben done!

Comment (moltenice) [reply plugins/airbrake/package.json]: Yeah I am directly using it here: https://github.com/backstage/backstage/pull/9605/files#diff-80249c8b6f20164606712f880c923987b3cf4bd7af18a8f4a216b5d6fb7a9f61R70

And I can make do by just implementing the type.

**Review by freben — APPROVED** [review]
Let's gooo!

**Pull request merged**